src/widgets/radio.jsx

```diff
@@ -19,6 +19,147 @@ var captureScratchpadTouchStart =
+    },
+
+    getDefaultProps: function() {
+        return {
+            checked: false,
+            classSet: {},
+            disabled: false,
+            showClue: false,
+            type: 'radio'
+        };
+    },
+
+    render: function() {
+        var classSet = _.clone(this.props.classSet);
+
+        classSet[ApiClassNames.RADIO.OPTION] = true;
+        classSet[ApiClassNames.RADIO.SELECTED] = this.props.checked;
+
+        return <div className={cx(classSet)}>
+            <span className="checkbox">
+                <input
+                    type={this.props.type}
+                    name={this.props.groupName}
+                    checked={this.props.checked}
+                    disabled={this.props.disabled}
+                    onClick={(e) => {
+                        // Avoid sending this to the parent
+                        e.stopPropagation();
+                    }}
+                    onChange={(e) => {
+                        this.props.onChecked(e.target.checked);
+                    }} />
+            </span>
+            {/* A pseudo-label. <label> is slightly broken on iOS,
+                so this works around that. Unfortunately, it is
+                simplest to just work around that everywhere. */}
+            <span className={
+                    ApiClassNames.RADIO.OPTION_CONTENT + " " +
+                    ApiClassNames.INTERACTIVE
+                }
+                style={{ cursor: "default" }}>
+                <div>
+                    {this.props.content}
+                </div>
+            </span>
+            {this.props.showClue &&
+                <div className="perseus-radio-clue">
+                    {this.props.clue}
+                </div>}
+        </div>
+    }
+});
+
+var ChoiceNoneAbove = React.createClass({
+    propTypes: {
+        showContent: React.PropTypes.bool
+    },
+
+    getDefaultProps: function() {
+        return {
+            showContent: true
+        };
+    },
+
+    render: function() {
+        var choiceProps = _.extend({}, this.props, {
+            classSet: { "none-of-above": true },
+            content: (this.props.showContent ?
+                this.props.content :
+                "None of the above"
+            ),
+        });
+
+        return <Choice {...choiceProps} />;
+    }
+});
+
+var ChoiceEditor = React.createClass({
+    propTypes: {
+        choice: React.PropTypes.shape,
+        showDelete: React.PropTypes.bool,
+        onClueChange: React.PropTypes.func,
+        onContentChange: React.PropTypes.func,
+        onDelete: React.PropTypes.func
+    },
+
+    render: function() {
+        var checkedClass = this.props.choice.correct ? "correct" : "incorrect";
+        var placeholder = "Type a choice here...";
+
+        if (this.props.choice.isNoneOfTheAbove) {
+            placeholder = this.props.choice.correct ?
+            "Type the answer to reveal to the user..." : "None of the above";
```

Comment:
This line should be indented over four spaces (and the "None of the above" will probably have to be on the next line to fit within 80 characters)